pkg/aws/aws.go

```diff
@@ -334,3 +342,37 @@ func convertGroup(g *autoscaling.Group) (*asg, error) {
 	}
 	return a, nil
 }
+
+func getDetachedInstances(svcEC2 *ec2.EC2, filter map[string]string) []*ec2.Instance {
+	detachedInstances := []*ec2.Instance{}
+	input := &ec2.DescribeInstancesInput{
+		Filters: []*ec2.Filter{},
+	}
+
+	for fkey, fvalue := range filter {
+		input.Filters = append(input.Filters, &ec2.Filter{
+			Name: aws.String(fmt.Sprintf("tag:%s", fkey)),
+			Values: []*string{
+				&fvalue,
+			},
+		})
+	}
+
```

Comment:
need to add the part to only get the running instances
`input.Filters = append(input.Filters, &ec2.Filter{
		Name: aws.String("instance-state-name"),
		Values: []*string{
			aws.String("running"),
		},
	})`